.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,29 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/docker-existing-dockerfile
+{
+	"name": "Existing Dockerfile",
+	"build": {
+		// Sets the run context to one level up instead of the .devcontainer folder.
+		"context": "..",
+		// Update the 'dockerFile' property if you aren't using the standard 'Dockerfile' filename.
+		"dockerfile": "../Dockerfile"
+	},
+	
+	//atfer container start finish, start tomcat service command
+	"postStartCommand": ["catalina.sh", "run"]
+
+	// Features to add to the dev container. More info: https://containers.dev/features.
+	// "features": {},
+
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	// "forwardPorts": [],
+
+	// Uncomment the next line to run commands after the container is created.
+	// "postCreateCommand": "cat /etc/os-release",
+
+	// Configure tool-specific properties.
+	// "customizations": {},
```

Comment:
I think you can add java related extensions here.

```json
"customizations": {
    "vscode": {
        "extensions": ["vscjava.vscode-java-pack", "vmware.vscode-boot-dev-pack"] // or more
    }
},
```

Reply:
May I ask, what functionalities can be achieved by installing java extensions in a devcontainer?
I can't think of any other extensions at the moment. Do you have any commonly used ones to recommend? 
Thank you very much egg bro.

Reply:
It's a java spring boot repo, so at least java and spring boot extensions are required I think.

Reply:
Oh, I understand. 
I'll look into this when I get home after work. Thanks! 
Maybe i'll adding Git control to manage these as well.

Reply:
> Maybe i'll adding Git control to manage these as well.

It should be good.